plugins/query-insights/src/main/java/org/opensearch/plugin/insights/core/service/TopQueriesService.java

```diff
@@ -80,7 +80,9 @@ public class TopQueriesService {
      * @param topNSize the top N size to set
      */
     public void setTopNSize(final int topNSize) {
-        this.topNSize = topNSize;
+        if (topNSize < QueryInsightsSettings.MAX_N_SIZE) {
+            this.topNSize = topNSize;
+        }
```

Comment:
By taking this approach we are failing silently and is not correct IMO. Better way is to invoke validation methods like `this.queryInsightsService.validateTopNSize(type, v)` before finishing the constructor of QueryInsightsListener around line 81